Sources/OktaIdx/IDXRemediationCollection.swift

```diff
@@ -0,0 +1,37 @@
+//
+// Copyright (c) 2021, Okta, Inc. and/or its affiliates. All rights reserved.
+// The Okta software accompanied by this notice is provided pursuant to the Apache License, Version 2.0 (the "License.")
+//
+// You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0.
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+// WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//
+// See the License for the specific language governing permissions and limitations under the License.
+//
+
+import Foundation
+
+extension IDXClient {
+    /// Represents a collection of remediation options.
+    @objc(IDXRemediationCollection)
+    public class RemediationCollection: NSObject {
+        /// Returns a remediation based on its string name.
+        @objc public subscript(name: String) -> Remediation? {
+            remediations.first { $0.name == name }
+        }
+        
+        /// Returns a remediation based on its type.
+        public subscript(type: Remediation.RemediationType) -> Remediation? {
+            remediations.first { $0.type == type }
+        }
+        
+        let remediations: [Remediation]
+        
+        init(remediations: [Remediation]?) {
+            self.remediations = remediations ?? []
```

Comment:
What is your opinion about avoiding Optional of collections on initialization, for instance like here?
Isn't it better to make just `init(remediations: [Remediation])`. Because setting collection to default value (e.g. empty array) may be confusing for end developer in some cases.

Reply:
The initializer is internal, so it won't be visible to the developer.  And allowing us to pass optional here drastically simplifies the parsing logic in the `v1` implementation section. It saves us from having to do a ton of unwrapping elsewhere.